### ISSUE TITLE
Parsing error with & in To address

### ISSUE THREAD

**Issue by dngray** [ProtonMail/proton-bridge#316]
## Expected Behavior

Email should be uploaded. When an address is like `something & something <something@example.com>` you'll get this error

## Current Behavior

An error like:

```
Failed to parse address   error="no viable alternative at input 'XXXX & '"
```

## Steps to Reproduce
1. Try to import an email over the bridge with an & in the To field.

## Version Information
3.0.4

## Context (Environment)
Using docker container [shenxn/protonmail-bridge-docker](https://github.com/shenxn/protonmail-bridge-docker) because this is most suitable for my needs.

On a side note there, it would be good if Proton had an official Docker container.

**Comment by dngray** [comment]
I also saw another one:
```
Failed to parse address     error="no viable alternative at input '\"Mail Delivery System <>\" <@>'"
```
In this case the From address was ``From: Mail Delivery System <>`

The email was a mail that was failed to deliver.

**Comment by dngray** [comment]
An another one I discovered was the user had used an incorrect address:

```
user@example.com]
```

(Obviously not a parseable email)

In Proton Mail it shows up as `To: Undisclosed Recipients`

The real email is only in the header.

**Comment by dngray** [comment]
> Interesting, your example with `&` in the address display name works fine with our legacy [go-rfc5322](https://github.com/protonmail/go-rfc5322) parser (used by bridge v2), but the [C++ rewrite](https://github.com/ProtonMail/gluon/tree/dev/internal/parser) (used by bridge v3 -- thanks for alpha testing!) doesn't like it. We'll take a look ASAP (internal reference: GODT-2185)

I'm about to pass about 12GB of email through the bridge as a test, to see what kind of issues might come up.

> Your example `"Mail Delivery System <>" <@>` is not a valid RFC5322 address, as the local-part and domain (which appear either side of the `@` in the angle brackets) must not be empty.

Yes, that particular example was one where a user had saved a mail delivery error in one of their folders.

I found another one that it fails to parse on:

```
Failed to parse address     error="no viable alternative at input 'XXXXX -'"
```

Seems in one particular case this user had "name" and then their organization after the hyphen.

**Comment by LBeernaertProton** [comment]
The issue has been found and addressed. It will be deployed in the next update. 

Thank you again for the detailed feedback.

**Comment by jameshoulahan** [comment]
@dngray the address parser has been updated in the [v3.0.6](https://github.com/ProtonMail/proton-bridge/releases/tag/v3.0.6) release to better handle display names. Would you mind giving it a go and reporting back your findings?

**Comment by dngray** [comment]
Seems these are fixed in the 3.0.6 branch I need to find out more information about that [above one](https://github.com/ProtonMail/proton-bridge/issues/316#issuecomment-1337157742) though:

```
Failed to parse address    error="mismatched input '<EOF>' expecting {',', '<', CFWS, QuotedText, EncodedAtom, Atom}"
```

**Comment by jameshoulahan** [comment]
Glad to hear the issue has been resolved! As such, I'll close this thread.

Regarding your last comment, it seems the message contains an empty address. This shouldn't prevent bridge from showing the message, we just log the error for completeness. If you manage to find the offending message and have any issues with the message as a result, please open a new GitHub issue and I'll look into it.
